src/crawlee/http_clients/_httpx.py

```diff
@@ -66,6 +66,11 @@ class HttpxHttpClient(BaseHttpClient):
 
     This client uses the `HTTPX` library to perform HTTP requests in crawlers (`BasicCrawler` subclasses)
     and to manage sessions, proxies, and error handling.
+
+    The client raises an `HttpStatusCodeError` when it encounters an error response, defined by default as any
```

Comment:
I think this should apply to any http client, right? Couldn't we just put this into the base client doc and link to it here?

Reply:
It should, but the implementation of such behavior is in the specific HTTP clients, not the base. If someone creates their own HTTP client, they are not required to implement it in the same way. As a result, the docstring may not be true.

Reply:
Valid point. In that case, we may want to pull this logic up into `BaseHttpClient` though. It seems unlikely you'd want to change this behavior, and at the same time, we don't want to make http-client-authors implement this over and over again.